add UseInboxContract option

blockchaindevsh opened the pull request:
This PR adds a `UseInboxContract` field to DeployConfig and RollupConfig so that this feature is only enabled when UseInboxContract is `true`, in order to prepare for becoming a superchain.

blockchaindevsh committed:
add UseInboxContract option

qizhou commented on op-node/rollup/derive/blob_data_source.go:
```diff
@@ -73,6 +73,22 @@ func (ds *BlobDataSource) Next(ctx context.Context) (eth.Data, error) {
 	return data, nil
 }
 
+func getTxSucceedIfUseInboxContract(ctx context.Context, useInboxContract bool, fetcher L1Fetcher, hash common.Hash) (txSucceeded map[common.Hash]bool, err error) {
```
Need some comments on this function?

blockchaindevsh replied:
Done, also optimized a bit by returning only successful ones.

blockchaindevsh committed:
1. add comment for getTxSucceedIfUseInboxContract
2. optimize getTxSucceedIfUseInboxContract by only including an entry for successful tx hash
